### DISCUSSION TITLE
Homebrew for a multi-user macos host

### DISCUSSION THREAD

**Discussion by tbrowder** [Homebrew/discussions#5480] [Writing Formulae/Casks] [answered]
How can I, as an administrator on a MacOS host, install homebrew for use by non-admin users?

**Comment by gromgit** [comment, marked as answer]
This is actually answered in the Homebrew FAQ, though in a [rather obscure way](https://docs.brew.sh/FAQ#why-does-homebrew-say-sudo-is-bad):

>If you need to run Homebrew in a multi-user environment, consider creating a separate user account specifically for use of Homebrew.

So basically:
1. Create an account just for Homebrew (say, `hb`)
2. Install Homebrew as user `hb` (if you have an existing installation, `sudo chown -R hb: $(brew --prefix)` should fix it right up without the need for reinstallation)
3. Add an entry to `/etc/sudoers` (better, create a new `/etc/sudoers.d/hb` file) with something like the following:
```
hb     ALL = (ALL) NOPASSWD: /opt/homebrew/bin/brew    # or the output of $(which brew)
```
4. Tell all your users to `sudo hb brew ...`, or write a simple wrapper script that dose the same thing.

**Reply by tbrowder** [reply]
Thank you very much for your help. That should do the trick.

**Reply by tpkee** [reply]
> This is actually answered in the Homebrew FAQ, though in a [rather obscure way](https://docs.brew.sh/FAQ#why-does-homebrew-say-sudo-is-bad):
> 
> > If you need to run Homebrew in a multi-user environment, consider creating a separate user account specifically for use of Homebrew.
> 
> So basically:
> 
> 1. Create an account just for Homebrew (say, `hb`)
> 2. Install Homebrew as user `hb` (if you have an existing installation, `sudo chown -R hb: $(brew --prefix)` should fix it right up without the need for reinstallation)
> 3. Add an entry to `/etc/sudoers` (better, create a new `/etc/sudoers.d/hb` file) with something like the following:
> 
> ```
> hb     ALL = (ALL) NOPASSWD: /opt/homebrew/bin/brew    # or the output of $(which brew)
> ```
> 
>  4. Tell all your users to `sudo hb brew ...`, or write a simple wrapper script that dose the same thing.

This doesn't seem to work as it still asks for pwd

**Reply by gromgit** [reply]
> This doesn't seem to work as it still asks for pwd

What `brew` command are you running that causes it to ask for a password?

**Reply by tpkee** [reply]
Just brew. If I try to execute it as `sudo username brew` it throws an error because username is not a command, so I use `-Hu`

**Reply by gromgit** [reply]
Yeah, sorry, forgot about the `-u` flag. `-H` may or may not be necessary, it's been a while since I last used a multi-user setup.

As for `brew` asking for a password, that should only happen (off the top of my head) if you:
* `sudo -u hb brew install` a cask on macOS _and_ the cask installer requires root, because of this recent change: https://github.com/Homebrew/brew/pull/17694, or
* `sudo brew ...`, because you're asking for root as yourself rather than `hb`

 If you're being asked for a password under any other circumstance, that may warrant some investigation.
